common/alerts.js

```diff
@@ -52,5 +52,21 @@
                 };
             }
         };
-    }]);
+    }])
+    .directive('compile', ['$compile', function ($compile) {
+      return function(scope, element, attrs) {
+        scope.$watch(
+          function(scope) {
+            return scope.$eval(attrs.compile);
+          },
+          function(value) {
+            element.html(value);
+            $compile(element.contents())(scope);
+          }
+       )};
+     }]).controller('alertController', function($scope, Session) {
```

Comment:
You don't need to define a controller here. You could have the `login` function definition to the `link` function of `alerts` directive.